WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/CopyMediaToAppStorageUseCase.kt

```diff
@@ -14,15 +14,18 @@ import javax.inject.Named
 @Reusable
 class CopyMediaToAppStorageUseCase @Inject constructor(
     private val mediaUtilsWrapper: MediaUtilsWrapper,
-    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
 ) {
     /*
    * Some media providers (eg. Google Photos) give us a limited access to media files just so we can copy them and then
    * they revoke the access. Copying these files must be performed on the UI thread, otherwise the access might be
    * revoked before the action completes. See https://github.com/wordpress-mobile/WordPress-Android/issues/5818
+   * they revoke the access. Copying these files must be performed within the context (Activity) that requested the
+   * files, otherwise the access might be revoked before the action completes.
+   * See https://github.com/wordpress-mobile/WordPress-Android/issues/5818
    */
     suspend fun copyFilesToAppStorageIfNecessary(uriList: List<Uri>): CopyMediaResult {
-        return withContext(mainDispatcher) {
+        return withContext(bgDispatcher) {
```

Comment:
AH! before we merge, can we add [this check](https://github.com/wordpress-mobile/WordPress-Android/pull/13352/files#diff-c85806f6820a41ec9a43ee9f4b74536efb93928240e47de7668f342813aeff33R23-R25) here?

```
  if (!mediaUtilsWrapper.isInMediaStore(mediaUri) &&
                                // don't copy existing local files
                                !mediaUtilsWrapper.isFile(mediaUri)) {
[...]
```

because once it's copied as a local file, it may be copied once more when added through this same use case being reused in other use cases such as `addNewMediaToEditorAsync` in `AddLocalMediaToPostUseCase` and such

lmk 🙏

Reply:
Thanks for the change in 1e9844f! Verified it works as expected 👍